lib/GenomeFileUtil/core/GenbankToGenome.py

```diff
@@ -473,38 +475,35 @@ def _validate_existing_assembly(self, assembly_ref, genome_obj):
         if len(unmatched_ids_md5s) > 0:
             raise ValueError(warnings["assembly_ref_diff_seq"].format(", ".join(unmatched_ids_md5s)))
 
-    def _load_contigs_for_and_validate_existing_assemblies(self, genome_objs):
-        ref2genome = {}
-        for genome_obj in genome_objs:
-            assembly_ref = genome_obj.use_existing_assembly
-            if assembly_ref:
-                contigs = Bio.SeqIO.parse(genome_obj.consolidated_file, "genbank")
-                extra_info = self._get_contigs_and_extra_info(
-                    contigs, genome_obj
-                )
-                self._validate_existing_assembly(
-                    assembly_ref, genome_obj
-                )
-                ref2genome[assembly_ref] = genome_obj
-                genome_obj.extra_info = extra_info
-                genome_obj.assembly_ref = assembly_ref
-                genome_obj.assembly_path = None
-
-        if ref2genome:
-            assembly_refs = list(ref2genome.keys())
-            assembly_objs_info = self._get_objects_info(assembly_refs)
-            for assembly_ref, assembly_info in zip(
-                assembly_refs, assembly_objs_info
-            ):
-                genome = ref2genome[assembly_ref]
-                self._set_gc_content_dna_size_and_md5(genome, assembly_info)
+    def _load_contigs(self, genome_obj):
+        contigs = Bio.SeqIO.parse(genome_obj.consolidated_file, "genbank")
+        extra_info = self._get_contigs_and_extra_info(contigs, genome_obj)
+        genome_obj.extra_info = extra_info
+
+    def _validate_existing_assemblies(self, genome_obj):
+        assembly_ref = genome_obj.use_existing_assembly
+        if assembly_ref:
+            self._validate_existing_assembly(assembly_ref, genome_obj)
+            genome_obj.assembly_ref = assembly_ref
+            genome_obj.assembly_path = None
+
+            assembly_objs_info = self._get_objects_info([assembly_ref])
```

Comment:
This used to be batched, but now it's getting called once per genome, which seems like the wrong way to go

Reply:
validate existing assembly step should be placed before _parse_genbank step, which is now in for loop.

Reply:
So run the loop through load contigs, validate the configs (including this line) in a batch, and then loop over the genome objects and parse the genbank file

Reply:
so you think is worth the effort?
1. https://github.com/kbaseapps/GenomeFileUtil/pull/208#discussion_r1480745129
2. Not failing early if some are invalid

Reply:
Oh, this is only for when an assembly ref is provided. Ok, fair enough then
:+1: